src/thunderbots/software/util/refbox_constants.cpp

```diff
@@ -25,6 +25,18 @@ static const std::map<RefboxGameState, std::string> refbox_game_state_names = {
     {RefboxGameState::BALL_PLACEMENT_THEM, "BALL_PLACEMENT_THEM"},
     {RefboxGameState::REFBOX_GAME_STATE_COUNT, "REFBOX_GAME_STATE_COUNT"}};
 
+std::string name(const RefboxGameState& state)
+{
+    if (refbox_game_state_names.find(state) != refbox_game_state_names.end())
+    {
+        return refbox_game_state_names.find(state)->second;
+    }
+    else
+    {
+        return "INVALID RefboxGameState";
```

Comment:
This seems like a pretty sub-ideal way to handle this case:
- the `RefboxGameState` isn't invalid, there just isn't an entry for it in the map
- we failed to find a name, but we're still successfully returning a string and relying on the user to see it propogated somewhere, trace it back here, and figure out the issue

I'd much rather we (in decreasing order of how nice I think the solution is):
- Refactor this enum to be a class hierarchy. So `GameState` would be an abstract class, and then `BallPlacementUs`, `BallPlacementThem`, `KickOffUs`, etc. would extend `GameState`. If you make each have a `getName()` function and make it pure virtual in `GameState`, we can enforce that every state must define it's own string
- return an optional
- throw an exception

Reply:
now throws an exception